applications: nrf_desktop: Miscellaneous documentation improvements (#10350)
nrfconnect/sdk-nrf: applications/nrf_desktop/README.rst

* Kconfig files - These reflect the software configuration.
  See :ref:`kconfig_tips_and_tricks` for information about how to configure them.

For information about differences between DTS and Kconfig, see :ref:`zephyr:dt_vs_kconfig`.

Application-specific Kconfig configuration
==========================================

The nRF Desktop introduces Kconfig options that can be used to simplify an application configuration.
These options can be used to select a device role and to automatically apply a default configuration suitable for the selected role.

.. note::
   Part of the default configuration is applied by modifying the default values of Kconfig options.

>    Keep in mind that changing configuration in menuconfig does not automatically adjust user-configurable values to the new defaults, so that the values must be updated manually.
>    See the section ``Stuck symbols in menuconfig and guiconfig`` on the :ref:`kconfig_tips_and_tricks` in the Zephyr documentation for details.

Reviewer: ```suggestion
   Changing configuration in menuconfig does not automatically adjust user-configurable values to the new defaults. So, you must update those values manually.
   For more information, see the Stuck symbols in menuconfig and guiconfig section on the :ref:`kconfig_tips_and_tricks` in the Zephyr documentation.
```

>    The default Kconfig option values are automatically updated if configuration change is applied directly in the configuration files.

Reviewer: ```suggestion
   The default Kconfig option values are automatically updated if configuration changes are applied directly in the configuration files.
```

The nRF Desktop application aligns the configuration with the nRF Desktop use case by overlaying Kconfig defaults and selecting or implying the required Kconfig options.
Among others, the Kconfig :ref:`app_event_manager` and :ref:`lib_caf` options are selected to ensure that they are enabled.

> The :ref:`CONFIG_DESKTOP_SETTINGS_LOADER <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_POWER_MANAGER <config_desktop_app_options>` are implied to enable respectively :ref:`nrf_desktop_settings_loader` and :ref:`nrf_desktop_power_manager`.

Reviewer: ```suggestion
The :ref:`CONFIG_DESKTOP_SETTINGS_LOADER <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_POWER_MANAGER <config_desktop_app_options>` are implied to enable the :ref:`nrf_desktop_settings_loader` and :ref:`nrf_desktop_power_manager` modules, respectively.
```

When adding a new board for the first time, focus on a single configuration.
Moreover, keep the default ``debug`` build type that the application is built with, and do not add any additional build type parameters.
The following procedure uses the gaming mouse configuration as an example.

> Zerphyr support for a board

Reviewer 2: Zephyr

     * The application uses two logical LEDs - one for the peers state, and one for the system state indication.
     * Each of the logical LEDs can have either one (monochromatic) or three color channels (RGB).
       Such color channel is a physical LED.
     * The module uses Zephyr's :ref:`zephyr:led_api` driver for setting the LED color.
       Zephyr's LED driver can use the implementation based on either GPIO or PWM (Pulse-Width Modulation).
       The hardware configuration is described through DTS.
       See the :ref:`caf_leds` configuration section for details.

#. Review the :ref:`nrf_desktop_hid_configuration`.

> #. By default nRF Desktop device enables Bluetooth connectivity support.

Reviewer: ```suggestion
#. By default, the nRF Desktop device enables Bluetooth connectivity support.
```
